### PR TITLE
[Search Block] Improvements to make just 1 query to the backend

### PR DESCRIPTION
**Pull request by erral**
- cypress tests to check that search block issues just 1 query to the backend
- change composing order to query the bakckend once

### CODE REVIEW

**Comment by ionlizarazu** [comment]
fixes #5219

File: src/components/manage/Blocks/Search/SearchBlockView.jsx [hunk 1]
```diff
@@ -106,4 +106,6 @@ export const SearchBlockViewComponent = compose(
   (Component) => React.memo(Component, blockPropsAreChanged),
 )(SearchBlockView);
 
-export default compose(withQueryString, withSearch())(SearchBlockViewComponent);
+export default compose(withSearch())(
+  compose(withQueryString)(SearchBlockViewComponent),
```
Comment (tiberiuichim) [inline src/components/manage/Blocks/Search/SearchBlockView.jsx:110]: I don't understand this diff. In effect, the execution order is the same, right?

File: src/components/manage/Blocks/Search/hocs/withSearch.jsx [hunk 2]
```diff
@@ -333,7 +333,8 @@ const withSearch = (options) => (WrappedComponent) => {
       setSearchData(
         getInitialState(data, facets, urlSearchText, id, sortOn, sortOrder),
       );
-    }, [deepFacets, facets, data, urlSearchText, id, sortOn, sortOrder]);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (tiberiuichim) [inline src/components/manage/Blocks/Search/hocs/withSearch.jsx:336]: maybe use JSON.parse(deepFacets) so you can avoid the eslint-disable line?

File: src/components/manage/Blocks/Search/SearchBlockView.jsx (hunk 1 above)
Comment (davisagli) [reply src/components/manage/Blocks/Search/SearchBlockView.jsx:110]: @tiberiuichim `compose` operates from right to left. So, before the change, withSearch() was applied first, but now it is applied second.

There's no need to use compose here; it could be written as withSearch()(withQueryString(SearchBlockViewComponent))

@ionlizarazu It would be helpful if you could explain a bit about how changing the order solves the problem.

Comment (ionlizarazu) [reply src/components/manage/Blocks/Search/SearchBlockView.jsx:110]: @tiberiuichim I found that this wasn't really the problem, but as @davisagli said, the order is different and it makes sense to me to execute first the withQueryString and after the withSearch.

I'll remove the composes to make it more readable.

File: src/components/manage/Blocks/Search/hocs/withSearch.jsx (hunk 2 above)
Comment (ionlizarazu) [reply src/components/manage/Blocks/Search/hocs/withSearch.jsx:336]: it makes sense, I also have to add a deepData to remove the eslint-disable line.

File: src/components/manage/Blocks/Search/SearchBlockView.jsx [hunk 3]
```diff
@@ -106,4 +106,4 @@ export const SearchBlockViewComponent = compose(
   (Component) => React.memo(Component, blockPropsAreChanged),
 )(SearchBlockView);
```
Comment (sneridagh) [inline src/components/manage/Blocks/Search/SearchBlockView.jsx:108]: Why did you remove the compose?

**Review by sneridagh — COMMENTED** [review]
Just one comment, the rest LGTM if @tiberiuichim approves.

**Comment by sneridagh** [comment]
Merging, since it seems it fixes the flaky test problem we have in main.

File: src/components/manage/Blocks/Search/SearchBlockView.jsx (hunk 3 above)
Comment (ionlizarazu) [reply src/components/manage/Blocks/Search/SearchBlockView.jsx:108]: https://github.com/plone/volto/pull/5340#discussion_r1375475109
